apps/src/StudioApp.js

```diff
@@ -914,6 +917,20 @@ StudioApp.prototype.addChangeHandler = function(newHandler) {
 };
 
 StudioApp.prototype.runChangeHandlers = function() {
+  if (
+    this.isRunning() &&
+    this.editDuringRunAlert === undefined &&
+    (this.isUsingBlockly() ||
+      this.getCode().trim() !== this.executingCode.trim())
+  ) {
+    // we trim the whitespace from the start and end of the code before comparing in droplet
+    // because droplet sometimes adds an extra newline when switching from block to code mode
+    this.editDuringRunAlert = this.displayWorkspaceAlert(
+      'warning',
+      <div>{msg.editDuringRunMessage()}</div>,
+      true
+    );
+  }
```

Comment:
This seems like it should be a `changeHandler` rather than being added at the beginning of this function. You can add it as a change handler using `addChangeHandler`